source/core/grids.cc

```diff
@@ -413,30 +413,78 @@ read_mesh_and_manifolds_for_stator_and_rotor(
 
   // Rotor triangulation
   Triangulation<dim> rotor_temp_tria;
-  attach_grid_to_triangulation(rotor_temp_tria, *mortar_parameters.rotor_mesh);
-  GridTools::rotate(mortar_parameters.rotor_mesh->rotation_angle,
-                    rotor_temp_tria);
+  attach_grid_to_triangulation(rotor_temp_tria, *mortar.rotor_mesh);
+  GridTools::rotate(mortar.rotor_mesh->rotation_angle, rotor_temp_tria);
+
+  // Get stator manifold ids without flat id
+  unsigned int stator_ids_no_flat = 0;
+  for (const auto &id : stator_temp_tria.get_manifold_ids())
+    {
+      if (id != numbers::flat_manifold_id)
+        stator_ids_no_flat++;
+    }
+
+  // Get rotor manifold ids without flat id
+  unsigned int rotor_ids_no_flat = 0;
+  for (const auto &id : rotor_temp_tria.get_manifold_ids())
+    {
+      if (id != numbers::flat_manifold_id)
+        rotor_ids_no_flat++;
+    }
+
+  // Faces at rotor-stator interface
+  unsigned int n_faces_rotor_interface  = 0;
+  unsigned int n_faces_stator_interface = 0;
 
-  // Check manifold IDs at rotor boundary
-  std::set<unsigned int> rotor_boundary_manifold_ids;
   // Shift rotor boundary IDs #
+  // Shift rotor face manifold IDs #
+  // Check number of faces at rotor interface with stator
   for (const auto &face : rotor_temp_tria.active_face_iterators())
-    if (face->at_boundary())
-      {
-        face->set_boundary_id(face->boundary_id() +
-                              stator_temp_tria.get_boundary_ids().size());
-        if (face->manifold_id() != numbers::flat_manifold_id)
-          rotor_boundary_manifold_ids.insert(face->manifold_id());
-      }
-
-  // Check manifold IDs at stator boundary
-  std::set<unsigned int> stator_boundary_manifold_ids;
+    {
+      if (face->at_boundary())
+        {
+          face->set_boundary_id(face->boundary_id() +
+                                stator_temp_tria.get_boundary_ids().size());
+          if (face->boundary_id() == mortar.rotor_boundary_id)
+            n_faces_rotor_interface++;
+        }
+      if (face->manifold_id() != numbers::flat_manifold_id)
+        face->set_manifold_id(face->manifold_id() + stator_ids_no_flat);
```

Comment:
We need to check if boundary and manifold ids of the rotor are contiguous.

Reply:
And for GMSH grid they will all be set at once in a single grid, so this should be fine.